### PR TITLE
Better handling of screensharing ice connection state errors.

### PR DESCRIPTION
**Pull request by MaximKhlobystov**
This PR add the code to detect ice connection state changes when screensharing. When the state becomes `failed`, toast notification with a non-generic error message is thrown.

### CODE REVIEW

File: bigbluebutton-html5/imports/api/screenshare/client/bridge/kurento.js [hunk 1]
```diff
@@ -14,6 +15,8 @@ const SCREENSHARE_VIDEO_TAG = 'screenshareVideo';
 
 const CHROME_EXTENSION_KEY = CHROME_CUSTOM_EXTENSION_KEY === 'KEY' ? CHROME_DEFAULT_EXTENSION_KEY : CHROME_CUSTOM_EXTENSION_KEY;
 
+const ICE_CONNECTION_FAILED = "ICE connection failed";
```
Comment (antobinary) [inline bigbluebutton-html5/imports/api/screenshare/client/bridge/kurento.js]: Unused

File: bigbluebutton-html5/client/compatibility/kurento-extension.js [hunk 2]
```diff
@@ -367,6 +367,14 @@ Kurento.prototype.startScreensharing = function () {
       return kurentoManager.exitScreenShare();
     };
   });
+  this.webRtcPeer.peerConnection.oniceconnectionstatechange = () => {
+    if(this.webRtcPeer) {
```
Comment (antobinary) [inline bigbluebutton-html5/client/compatibility/kurento-extension.js]: Are we able to simply put these 2 new handlers to https://github.com/bigbluebutton/bigbluebutton/blob/242200ac915f67a4dc32a5f6d30b36faf74ca6fe/bigbluebutton-client/resources/prod/lib/kurento-extension.js to handle failed state in Flash client presenter?

Comment (prlanzarin) [reply bigbluebutton-html5/client/compatibility/kurento-extension.js]: Yes, I'll be porting that to the flash script with some alterations along with SFU error handling, but in a separate PR.

File: bigbluebutton-html5/client/compatibility/kurento-extension.js [hunk 3]
```diff
@@ -367,15 +362,23 @@ Kurento.prototype.startScreensharing = function () {
       return kurentoManager.exitScreenShare();
     };
   });
+  this.webRtcPeer.peerConnection.oniceconnectionstatechange = () => {
+    if (this.webRtcPeer) {
+      const connectionState = this.webRtcPeer.peerConnection.iceConnectionState;
+      if (connectionState === 'failed') {
```
Comment (prlanzarin) [inline bigbluebutton-html5/client/compatibility/kurento-extension.js]: Should add a `closed` listener for both presenter and viewer peers, same thing as with webcams.

**Pull request merged**